### ISSUE TITLE
Pyccel's Code Translation Error: Converts List Index l[o] to l[o,::]

### ISSUE THREAD

**Issue by mustapha-belbiad** [pyccel/pyccel#1732]
### When using Pyccel to process the Python code:

```python
l = [[1, 2, 3], [57, 8, 9]]
l[0]
```
**Pyccel generates incorrect code that alters the intended behavior. 
Instead of accessing the single element at index 0 of the first sublist `l[0]`, the generated code attempts to access a slice of the entire first sublist `l[0,::]`**

 **the Pyccel-generated code:**
```python
l = [[1, 2, 3], [57, 8, 9]]
l[0,::]
```
**However, it's crucial to acknowledge that the syntax lis[0,::] is not valid for list indexing in Python. Typically, list indexing allows accessing elements or sublists using integers (e.g., l[0]) or slices (e.g., l[1:3])**

### Error details
During the execution of the Pyccel-generated code, I get the following error message:
```bash/fortran/c
Traceback (most recent call last):
  File "/home/mbelbiad/Desktop/clear_method/zz/test.py", line 3, in <module>
    Dummy_0000 = lis[0,::]
                 ~~~^^^^^^
TypeError: list indices must be integers or slices, not tuple
```

**Comment by EmilyBourne** [comment]
Does this work for tuples?

**Comment by mustapha-belbiad** [comment]
> Does this work for tuples?

the Pyccel-generated code for tuple:
```python
lis = ([1, 2, 3, 4], [3, 6, 788, 8], [66, 8, 0, 2])

Dummy_0000 = lis[1][::]
```

**Comment by EmilyBourne** [comment]
Interesting. We must be doing something different for tuples then as the generated code is valid for tuples. I wonder if this is related to the rank? If so then this issue is related to #1722

**Comment by EmilyBourne** [comment]
The fix for tuples is here:
https://github.com/pyccel/pyccel/blob/666aa640c8cf4e6253a92ff5f9a2f51534ac7bdf/pyccel/codegen/printing/pycode.py#L295-L296

If we need this fix for a test then we could also add lists as a temporary fix but ideally this should be fixed via the improvements in #1722
